rtf_vis_tool/src/Components/Bundle_Adj_PC.js

```diff
@@ -0,0 +1,114 @@
+/* Component that renders the Point Cloud and Camera Frustums before Bundle Adjustment. Spawned 
+once the 'SfMData' node is clicked.
+
+Author: Adi Singh
+*/
+import React, {useEffect, useState} from "react";
+
+// Third-Party Package Imports.
+import {Canvas} from "react-three-fiber"; // Used to render canvas containing 3D elements
+
+// Local Imports.
+import CoordinateGrid from './CoordinateGrid';
+import OrbitControlsComponent from './OrbitControlsComponent';
+import PointMesh from './PointMesh';
+import PointSizeSlider from './PointSizeSlider';
+import '../stylesheets/Bundle_Adj_PC.css';
+
+const Bundle_Adj_PC = (props) => {
+    // Variables to store the point cloud information and toggle the coordinate grid display.
+    const [pointCloudRaw, setPointCloudRaw] = useState([]);
+    const [pointCloudJSX, setPointCloudJSX] = useState([]);
+    const [showCoordGrid, setShowCoordGrid] = useState(true);
+
+    // Point size defined by radius, initialized as 0.15.
+    const [pointRadius, setPointRadius] = useState(0.15);
+    const pointSizeArr = [pointRadius]; 
+
+    // Render points3D.txt from COLMAP ba_input directory.
+    useEffect(() => {
+        // Fetch the COLMAP file from the public directory.
+        fetch('results/ba_input/points3D.txt')
+            .then((response) => {
+                return response.text();
+            })
+            .then((data) => loadCOLMAPPointCloud(data))
+    }, []);
+
+    /*  Accepts the raw COLMAP points3D.txt file and converts it into an array of JSX formatted
+        PointMeshes (which are then rendered on screen).
+    */ 
+    const loadCOLMAPPointCloud = (data) => {
+        const arrStringPoints = data.split('\n');
+        var finalPointsJSX = [];
+                
+        // Remove the first 3 commented lines of points3D.txt.
+        for (var i = 0; i < 3; i++) {
+            arrStringPoints.shift();
+        }
+
+        const arrNumPoints = arrStringPoints.map(point => point.split(" ").map(Number));
+        setPointCloudRaw(arrNumPoints);
+
+        // Loop through array. convert strings to numbers. Append to final point cloud.
+        for (var index = 0; index < arrNumPoints.length; index += 1) {
+            var pointArr = arrNumPoints[index];
+                    
+            finalPointsJSX.push(
+                <PointMesh  
+                    position={[pointArr[1], pointArr[2], pointArr[3]]}  
+                    color={`rgb(${pointArr[4]}, ${pointArr[5]}, ${pointArr[6]})`} 
+                    size={pointSizeArr}/>
+                );
+        }
+        setPointCloudJSX(finalPointsJSX);
+    }
+
+    /* Updates the radius of all points within a point cloud.
+       Called everytime the react slider input is interacted with.
+    */
+    const updatePointSizes = (radius) => {
+        var finalPointsJSX = [];
+        for (var i = 0; i < pointCloudRaw.length; i += 1) {
+            var pointArr = pointCloudRaw[i];
```

Comment:
is `pointCloudRaw` a global variable?

Reply:
maybe an inline comment that `pointCloudRaw` is (N x 6) array, with first 3 columns as (x,y,z) and last 3 columns as (R,G,B) would be helpful

Reply:
so `pointCloudRaw` is a state variable, which is just a React term for saying its only accessible within the `Bundle_Adj_PC` component. So I'd think of it more as a local variable. Anything that is defined like `const [pointCloudRaw, setPointCloudRaw] = useState([]);` is considered a react state which is locally defined to the component